pkg/operation/botanist/addons/nginxingress/nginxingress.go

```diff
@@ -0,0 +1,699 @@
+	// ConfigData contains the configuration details for the nginx-ingress controller.
+	ConfigData map[string]string
+	// KubeAPIServerHost is the host of the kube-apiserver.
+	KubeAPIServerHost string
+	// LoadBalancerSourceRanges is list of allowed IP sources for NginxIngress.
+	LoadBalancerSourceRanges []string
+	// ExternalTrafficPolicy controls the `.spec.externalTrafficPolicy` value of the load balancer `Service`
+	// exposing the nginx-ingress.
+	ExternalTrafficPolicy corev1.ServiceExternalTrafficPolicyType
+	// VPAEnabled marks whether VerticalPodAutoscaler is enabled for the shoot.
+	VPAEnabled bool
+	// PSPDisabled marks whether the PodSecurityPolicy admission plugin is disabled.
+	PSPDisabled bool
+}
+
+// New creates a new instance of DeployWaiter for nginx-ingress
+func New(
+	client client.Client,
+	namespace string,
+	values Values,
+) component.DeployWaiter {
+	return &nginxIngress{
+		client:    client,
+		namespace: namespace,
+		values:    values,
+	}
+}
+
+type nginxIngress struct {
+	client    client.Client
+	namespace string
+	values    Values
+}
+
+func (n *nginxIngress) Deploy(ctx context.Context) error {
+	data, err := n.computeResourcesData()
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForShoot(ctx, n.client, n.namespace, ManagedResourceName, managedresources.LabelValueGardener, false, data)
+}
+
+func (n *nginxIngress) Destroy(ctx context.Context) error {
+	return managedresources.DeleteForShoot(ctx, n.client, n.namespace, ManagedResourceName)
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (n *nginxIngress) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, n.client, n.namespace, ManagedResourceName)
+}
+
+func (n *nginxIngress) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, n.client, n.namespace, ManagedResourceName)
+}
+
+func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
+	var (
+		healthProbePort = intstr.FromInt(10254)
+
+		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		configMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: configMapName,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+					labelKeyComponent:         labelValueController,
+				},
+				Namespace: n.namespace,
+			},
+			Data: n.values.ConfigData,
+		}
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceAccountName,
+				Namespace: n.namespace,
```

Comment:
```suggestion
				Namespace: addonNamespace,
```
nginx-ingress-addon should be deployed in `kube-system` namespace of the shoot.